product_barcode_generator/models/product.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+###############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (C) 2013 Julius Network Solutions SARL <contact@julius.fr>
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+###############################################################################
+
+from openerp import api, models, fields, _
+from openerp import exceptions
+
+
+def isodd(x):
+    return bool(x % 2)
+
+
+class ProductCategory(models.Model):
+    _inherit = 'product.category'
+
+    ean_sequence_id = fields.Many2one('ir.sequence', string='Ean sequence')
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    ean_sequence_id = fields.Many2one('ir.sequence', string='Ean sequence')
+
+    @api.model
```

Comment:
It can't be `@api.one` if there's a return. It should be `@api.multi` with `self.ensure_one()` sentence and taking self as the product.